tests/unit_tests/shared/test_feature_toggling.py

```diff
@@ -0,0 +1,69 @@
+import os
+from unittest.mock import Mock
+
+import pytest
+
+import ert.__main__
+from ert.__main__ import ert_parser
+from ert.cli import TEST_RUN_MODE
+from ert.shared.feature_toggling import FeatureToggling
+
+
+@pytest.fixture(autouse=True)
+def mocked_valid_file(monkeypatch):
+    monkeypatch.setattr(ert.__main__, "valid_file", Mock(return_value=""))
+
+
+def test_feature_toggling_from_env(monkeypatch):
+    envs = {"ERT_ENABLE_SCHEDULER": True}
+    monkeypatch.setattr(os, "environ", envs)
+
+    assert not FeatureToggling.is_enabled("scheduler")
+
+    FeatureToggling.update_from_environment()
+
+    assert FeatureToggling.is_enabled("scheduler")
+
+
+def test_feature_toggling_from_args():
+    parsed = ert_parser(
+        None,
+        [
+            TEST_RUN_MODE,
+            "--enable-scheduler",
+            "--enable-new-storage",
+            "not_a_real_config.ert",
+        ],
+    )
+
+    assert not FeatureToggling.is_enabled("scheduler")
+    assert not FeatureToggling.is_enabled("new-storage")
+
+    FeatureToggling.update_from_args(parsed)
+
+    assert FeatureToggling.is_enabled("scheduler")
+    assert FeatureToggling.is_enabled("new-storage")
+
+
+def test_feature_toggling_both(monkeypatch):
+    envs = {"ERT_ENABLE_SCHEDULER": False}
```

Comment:
Same comment about `monkeypatch.setenv`. Also, `False` is not a valid value (`os.environ` is `str -> str` mapping)

Reply:
fixed